clientconn.go

```diff
@@ -95,6 +96,14 @@ func WithTimeout(d time.Duration) DialOption {
 	}
 }
 
+// WithNetwork returns a DialOption that specifies the network on which
+// the connection will be established.
+func WithNetwork(network string) DialOption {
```

Comment:
On Fri, Apr 17, 2015 at 3:44 PM, David Symonds <notifications@github.com>
wrote:

> In clientconn.go
> <https://github.com/grpc/grpc-go/pull/169#discussion_r28635771>:
>
> > @@ -95,6 +96,14 @@ func WithTimeout(d time.Duration) DialOption {
> >  	}
> >  }
> >
> > +// WithNetwork returns a DialOption that specifies the network on which
> > +// the connection will be established.
> > +func WithNetwork(network string) DialOption {
>
> This feels too specific. It doesn't help this support any other address
> types either. Why don't we simply accept a dialer option that takes
> complete responsibility for turning the address into a net.Conn?
>
I do not get it. Can you elaborate? What is "a dialer option"? Are you
proposing to have our own struct to take complete responsibility for
turning the addr to a net.Conn?

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/169/files#r28635771>.
>